### PR TITLE
Fetch override list from ThemeDB

### PR DESCRIPTION
**Pull request by KoBeWi**
Part of #82402
Theme override list now uses ThemeDB binds instead of default theme.

### CODE REVIEW

**Comment by YuriSizov** [comment]
> Unfortunately the order of categories is shuffled depending on the order of binding of the properties. The best fix is probably enforcing binding order in `_bind_methods()` of Control classes.

I think we shouldn't rely on the declaration order. Instead when we bind/register theme items, we can sort it then. This won't be free, but it beats doing it on read.

However, the order of the properties themselves within each type should follow the declaration order, or at least the order returned by `ThemeDB` so it stays consistent across the surfaces where we display it.

**Comment by KoBeWi** [comment]
Ok I added a secondary HashMap that keeps bind items in a sorted list.

File: scene/theme/theme_db.cpp [hunk 1]
```diff
@@ -435,6 +442,7 @@ ThemeDB *ThemeDB::get_singleton() {
 
 ThemeDB::ThemeDB() {
 	singleton = this;
+	MessageQueue::get_singleton()->push_callable(callable_mp(this, &ThemeDB::_sort_theme_items));
```
Comment (YuriSizov) [inline scene/theme/theme_db.cpp:445]: Looks like this line causes a crash in unit tests on CI on every platform.

File: scene/theme/theme_db.cpp [hunk 2]
```diff
@@ -381,25 +383,30 @@ void ThemeDB::get_class_own_items(const StringName &p_class_name, List<ThemeItem
...
 }
 
-// Object methods.
+void ThemeDB::_sort_theme_items() {
+	for (KeyValue<StringName, List<ThemeDB::ThemeItemBind>> &E : theme_item_binds_list) {
+		E.value.sort_custom<ThemeItemBind::SortByType>();
+	}
+}
 
+// Object methods.
```
Comment (YuriSizov) [inline scene/theme/theme_db.cpp:409]: There was an empty line here, btw, to mark a section. It's not a comment for this specific method 🙃

File: scene/theme/theme_db.cpp (hunk 1 above)
Comment (YuriSizov) [reply scene/theme/theme_db.cpp:445]: Are you trying to call it just once for all registration calls by delaying it like that? If so, I'm not sure it's going to be reliable, but we can give it a go, if the crash is fixable.

**Review by YuriSizov — CHANGES_REQUESTED** [review]
While it works, there is an issue with the implementation which doesn't account for inherited properties. See https://github.com/godotengine/godot/pull/84763#pullrequestreview-1818721652. This is something that needs to be addressed first, and the crash needs investigating too, of course.

File: scene/theme/theme_db.cpp (hunk 1 above)
Comment (KoBeWi) [reply scene/theme/theme_db.cpp:445]: For better reliability I could add `queue_sort_items()` to call when an item is added, but currently adding is done only once, during initialization, so it's not needed.

**Review by YuriSizov — APPROVED** [review]
I now have my doubts about keeping the declaration order within each type, but this can be changed later. At least everything will be consistent once we use the same API everywhere.

The method should probably be renamed from `get_class_own_items` to `get_class_items`. I believe I named it with `own` to signify that it only returns its own defined items, nothing inherited. With the new flag, this is no longer true.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**